enhancements/olm/olm-managed-operator-metrics.md

```diff
@@ -0,0 +1,145 @@
+### Non-Goals
+
+- Provide generic metrics about OLM managed operators
+
+## Proposal
+
+### OLM support for new object in bundles
+
+The bundle object currently has the following format:
+
+```bash
+$ tree
+bundle
+    ├── manifests
+    │   ├── example.crd.yaml
+    │   ├── example.csv.yaml
+    │   └── example.rbac.yaml
+    └── metadata
+        └── annotations.yaml
+```
+
+Notice that OLM understands how to create and manage CSVs, CRDs, and RBAC files. This enhancement proposes expanding this feature to support `Namespaces`, `ServiceMonitors` and `PrometheusRules` objects:
+
+```bash
+$ tree
+bundle
+    ├── manifests
+    │   ├── example.crd.yaml
+    │   ├── example.csv.yaml
+    │   ├── example.ns.yaml
+    │   ├── example.prometheusrule.yaml
+    │   ├── example.rbac.yaml
+    │   └── example.servicemonitor.yaml
+    └── metadata
+        └── annotations.yaml
+```
+
+With this change, OLM will need to be updated to manage the `Namespace`, `ServiceMonitor` and `PrometheusRule` objects through the lifecycle of an operator.
+
+#### Namespace Support
+
+As OLM gains support for `Namespace` objects, The `ClusterServiceVersion` will be updated to include an optional `NamespaceRequirements` object field. The `NamespaceRequirements` will look like the following:
+
+```golang
+type NamespaceRequirements struct {
+  //+optional
+  // name represents where the operator must be deployed.
+  name string
+
+  //+optional
+  // labels are labels that must be present on the given namespace.
+  labels []string
+}
+```
+
+>Note: If `NamespaceRequirements.namespace` is set, the UI will only show the specified namespace as an install option.
+
+#### ServiceMonitor and PrometheusRule Support
+
+As OLM gains support for these Prometheus objects, there is a chance that installed operators may overload the Prometheus instance. It is important that a Cluster Admin is alerted that an operator will report metrics prior to deploying the operator. As such, we propose adding a `monitoringSupported` boolean field to the `ClusterServiceVersion` object. OLM will not install either of the Prometheus object unless the `monitoringSupported` field is set to true. Additionally, Cluster Admins will need to manually the operator's `InstallPlans` in the following scenarios:
+
+- A fresh install for this operator is being created where this field is set to `true`
+- An existing operator had this field set to `false` and is being upgraded to a new version that sets this field to `true`
+
+> Note: The `InstallPlan` will include a helpful message for Cluster Admins if either of the situations above are encountered.
+
+### OLM Telemeter Support
+
+#### Telemeter Requirements
+
+Once OLM is capable of managing `ServiceMonitor` and `PrometheusRule` objects along with the lifecycle of an operator, OLM must allow Red Hat owned operators to take advantage of Telemeter. Before moving forward, let's review the requirements that must be met for Telemeter to scrape an application's metrics:
```

Comment:
These are not the only requirements, it also needs to be whitelisted via telemeter. You have not detailed how to handle that part?

Reply:
+1

Have we thought about, instead of whitelisting, just marking the `redhat-operators` catalog as special? Have we ever considered catalog specific features before? What have we done previously?

Reply:
Whitelisting via telemeter is a different process correct? 

I agree we need to change the wording here s/Telemeter/OpenShift Monitoring correct and it is more clear? What we really are attempting to enable is an operator can be scraped by the OpenShift Monitoring IIUC WDYT @LiliC ?

Reply:
@shawn-hurley  is correct here - this highlights getting scraped by the monitoring-prometheus instance.

Reply:
Yes, the wording the proposal confused me. Lets change that, as per @shawn-hurley suggestions. 👍 

As for @kevinrizza I left some thoughts below, but lets discuss this in meeting/email/outside of PR then. See https://github.com/openshift/enhancements/pull/152#issuecomment-565443956

Reply:
I attempted to make it more apparent that we are integrating with the `OpenShift Monitoring Prometheus Operator` in the latest version of the PR.